### DISCUSSION TITLE
Remote Error: no Display

### DISCUSSION THREAD

**Discussion by medoer** [rustdesk/rustdesk#6950] [General]
### Bug Description

RustDesk Client Show me a "Remote Error: no Display" confirm dialog, when I remove the remote computer's HDMI。 But It works well when the remote computer connects with the display, and another remote computer works well when removing the HDMI.
The error machine "uname -a "  message: Linux xxx-Z790-UD 6.2.0-33-generic #33~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Sep  7 10:33:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
The work well one: Linux xxx 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### How to Reproduce

Remove the HDMI from the remote computer.

### Expected Behavior

Working when removing the HDMI.

### Operating system(s) on local side and remote side

macos mojave ->  22.04.1-Ubuntu

### RustDesk Version(s) on local side and remote side

1.2.3

### Screenshots

<img width="1440" alt="image" src="https://github.com/rustdesk/rustdesk/assets/20471718/106d4c8d-b7fc-4b40-9c51-09bfa4ab1ec9">

**Comment by Joaquim3** [comment]
I have the same issue. Apparently, this has been discussed previously [https://github.com/rustdesk/rustdesk/issues/59].

I my case, I was not satisfied with the proposed solutions. I don't want to use XRDP.

I have 2 computers with LinuxMint each. 
I have only 1 screen for both.
Temporary Solution -> I use only 1 HDMI adapter for both (1 screen -> 2 computers).

_But I'm anyway very Thankfull to the Developpers. I know this is not that easy._

**Comment by xlionjuan** [comment]
Buy HDMI dummy is the easiest solution.

**Reply by MLGith** [reply]
HDMI dummy did not work Zorin 18

**Comment by csvke** [comment]
For people who prefer not to use a HDMI dummy (or physically cannot put one in), a workaround is to use `xserver-xorg-video-dummy`

It creates a dummy device, monitor and screen running in your host so RustDesk can connect to the display created.

```
sudo apt-get update
sudo apt-get install xserver-xorg-video-dummy
```

```
sudo nano /etc/X11/xorg.conf.d/10-dummy.conf
```
This is an example configuration file.
```
Section "Device"
  Identifier "Dummy Device"
  Driver "dummy"
  VideoRam 256000
EndSection

Section "Monitor"
  Identifier "Dummy Monitor"
  HorizSync 28.0-80.0
  VertRefresh 48.0-75.0
  Modeline "1920x1080"  172.80  1920 2040 2248 2576  1080 1081 1084 1118 -hsync +vsync
EndSection

Section "Screen"
  Identifier "Dummy Screen"
  Monitor "Dummy Monitor"
  Device "Dummy Device"
  DefaultDepth 24
  SubSection "Display"
    Depth 24
    Modes "1920x1080"
  EndSubSection
EndSection

Section "ServerLayout"
  Identifier "Dummy Layout"
  Screen "Dummy Screen"
EndSection
```
`sudo reboot`

You can make changes to the monitor and screen resolutions and other parameters according to the screen of the client you use. 

Depending on which distribution of linux you use, the exact steps may vary but this has been tested in Ubuntu 24.04 Armbian_community 24.11.0-trunk.202 noble 

@rustdesk maybe have this documented in FAQ? The whole process needs some polishing but I read in another discussion here that there's no plans for you folks to add functionalities of creating a virtual display. I suppose a lot of the customers' use case would be to use RustDesk to connect to headless linux.

**Reply by sobermh** [reply]
it works good!!!!

**Reply by Ipherc** [reply]
It works. Thank you

Debian GNU/Linux 12 bookworm (x86-64)
Linux Kernel 6.1.0-33-amd64

**Reply by WindSlashKing** [reply]
It didn't work for me. I'm on Linux Mint. It made my physical display not work and there is no virtual one.

**Reply by xlionjuan** [reply]
> It didn't work for me. I'm on Linux Mint. It made my physical display not work and there is no virtual one.

So you need to enable SSH first.

**Reply by H-goto16** [reply]
Thanks!

**Reply by dakeeper** [reply]
Works like a charm.
Used it on a Rpi 4b with Kali 2025.4
Thank you!

**Reply by MLGith** [reply]
Zorin 18, after applying this file, Zorin starts but goes to a black screen, nothing works and I have no idea how to undo it, I'm screwed.

**Reply by xlionjuan** [reply]
Enter emergency mode to remove that file

**Reply by xlionjuan** [reply]
Enable SSH so you could run commands remotely

**Reply by MLGith** [reply]
> Enter emergency mode to remove that file

Thanks, I google how to enter emergency mode in Zorin and I found a couple of articles, finally I was able to do it.

**Comment by MLGith** [comment]
Sorry I need some help:

I am using Zorin Linux, and I do not have 

etc/X11/xorg.conf.d

and apparently there is no way to create that folder.. How can I continue?

Thanks for your help.

I am barely learning Linux, I have almost 0 knowledge, and I am moving from Windows.

**Reply by xlionjuan** [reply]
> Sorry I need some help:
> 
> I am using Zorin Linux, and I do not have 
> 
> etc/X11/xorg.conf.d
> 
> and apparently there is no way to create that folder.. How can I continue?
> 
> Thanks for your help.
> 
> I am barely learning Linux, I have almost 0 knowledge, and I am moving from Windows.


```
sudo mkdir -p /etc/X11/xorg.conf.d
```

**Reply by H-goto16** [reply]
I think mkdir cannot create the file.

```
sudo mkdir -p /etc/X11/
sudo touch /etc/X11/xorg.conf.d
```

is needed.

**Reply by MLGith** [reply]
Sorry the result was a file not a folder, xorg.conf.d if I am not wrong is supposed to be a folder and inside of that folder will contain the file called 10-dummy.conf, this is based on instructions at the beginning of this article:
sudo nano /etc/X11/xorg.conf.d/10-dummy.conf

**Reply by xlionjuan** [reply]
> I think mkdir cannot create the file.
> 
> ```
> sudo mkdir -p /etc/X11/
> sudo touch /etc/X11/xorg.conf.d
> ```
> 
> is needed.

??
conf.d's d is DIRECTORY

**Reply by MLGith** [reply]
yes there is supposed to be a file inside that folder: "10-dummy.conf" that file contains all the necessary instructions 

Thanks.

**Reply by H-goto16** [reply]
I completely misunderstood. I'm very sorry.

**Reply by MLGith** [reply]
Anyone know how can I create the folder xorg.conf.d under /etc/X11/
I have not been able to find any solution.

I AM NOT an expert on Linux I am a barely starting, I am using Zorin Linux V 16.

**Reply by KorewaLidesu** [reply]
> Anyone know how can I create the folder xorg.conf.d under /etc/X11/ I have not been able to find any solution.
> 
> I AM NOT an expert on Linux I am a barely starting, I am using Zorin Linux V 16.

`sudo mkdir /etc/X11/xorg.conf.d`?

**Reply by MLGith** [reply]
It works, now I'm going to try and see if RustDesk works.

**Comment by adityabhalsod** [comment]
I think RustDesk needs to handle this automatically.
